src/main.cpp

```diff
@@ -932,12 +932,13 @@ int main(int argc, char**argv)
 
   HssCacheTask::configure_health_checker(hc);
 
-  HttpConnection* http = new HttpConnection(options.sprout_http_name,
-                                            false,
-                                            http_resolver,
-                                            SASEvent::HttpLogLevel::PROTOCOL,
-                                            NULL);
-  SproutConnection* sprout_conn = new SproutConnection(http);
+  HttpClient* http_client = new HttpClient(false,
+                                           http_resolver,
+                                           SASEvent::HttpLogLevel::PROTOCOL,
+                                           nullptr);
+  HttpConnection* http_conn = new HttpConnection(options.sprout_http_name,
+                                                 http_client);
+  SproutConnection* sprout_conn = new SproutConnection(http_conn);
```

Comment:
Just had to look, as i was going to comment below that we need to delete the httpconnection as well, but a SproutConnection actually takes ownership of the connection and deletes it. Should we be updating it so that is not the case, bringing this more in line with e.g. ChronosConnection? 
Sadly, i've also just looked at hssconnection in Sprout, and that appears to create it's own connection and client, and delete them both as well, so we have no consistency across these connection classes; though it feels like at least having the creator also delete would be a good step forward

Reply:
Hmm, yeah, I wasn't sure about this at the time, but I think I agree that the fact that the creator doesn't also `delete` it is a bit untidy, as it's not obvious the SproutConnection is taking ownership. I'll change it so SproutConnection doesn't own the HttpConnection